File: 1/python/main.py
numbers = [3, 2, 4]
target = 6


def solve(target_sum, nums):
    nums_map = {}
    for i, num in enumerate(nums):
        remainder = target_sum - num
        if remainder in nums_map.keys():
            return nums_map[remainder], i
        else:
            nums_map[num] = i


print(solve(target, numbers))

File: 1/rust/src/main.rs
use std::collections::HashMap;

fn main() {
    let target = 6;
    let nums = vec![3, 2, 4];
    println!("{:?}", solve(target, nums));
}

fn solve(target: i32, nums: Vec<i32>) -> Vec<i32>  {
    let mut nums_map = HashMap::new();
    let mut answer: Vec<i32> = vec![];
    for (i, num) in nums.iter().enumerate() {
        let value = target - num;
        if nums_map.contains_key(&value) {
            answer.push(nums_map[&value]);
            answer.push(i.try_into().unwrap());
            break;
        } else {
            nums_map.insert(num, i.try_into().unwrap());
        }
    }
    answer
}